jedi/api/__init__.py

```diff
@@ -108,7 +110,11 @@ def __init__(self, source=None, line=None, column=None, path=None,
         self._user_context = UserContext(self.source, self._pos)
         self._parser = UserContextParser(self._grammar, self.source, path,
                                          self._pos, self._user_context)
-        self._evaluator = Evaluator(self._grammar)
+        if sys_path is None:
+            venv = os.getenv('VIRTUAL_ENV')
+            if venv:
+                sys_path = list(get_venv_path(venv)) + sys.path
```

Comment:
Yes, if Jedi is started in a virtualenv, it will get a proper `sys.path`.

As of now, I'm restarting the RPC server process, but in general it would be nice to be able to use one server for completions in different projects/virtualenvs.

FWIW, I think it is the right decision to start up some sort of a wrapper in the target environment, for stability and flexibility, but I don't think it's necessary to over-engineer it. All it needs to do is to provide some bare-minimum-RPC without dependencies, so that jedi can simply execute the script without forcing the user to install anything. It also doesn't need to be asynchronous, but being able to cancel an operation would be nice. Maybe even something as simple as json-over-stdin/stdout with a couple of threads (to be able to send and receive cancellation signals) would work.